articles/howtos/create-new-decorator.md

```diff
@@ -0,0 +1,222 @@
+---
+uid: CreateNewDecorator
+---
+# How can I create a new Decorator?
+
+Click [here](xref:Decorator) to find out more about Decorators.
+
+This how-to will serve as an example of how to create a new Decorator that will inject code at runtime for a Template that has been setup for it.
+
+This how-to will build up from the [previous](xref:MakeTemplateDecoratorReady) how-to, which showed how to make a Template ready for Decorators. Please interpolate on your end what is written here to make the necessary adjustments on what you're trying to achieve.
+
+## Create a new Decorator
+
+Typically you would create a new Decorator in a new Intent Architect Application which will reference a Template from another Module or Intent Architect Application. As such we will be doing that.
+
+![Startup](images/create-new-decorator/MyDecoratorModuleStartup.png)
+
+Go ahead and create your Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.
+
+Once the generation is done, create a new Decorator called `StaticFileServerDecorator`.
+
+![New Decroator](images/create-new-decorator/NewDecorator.png)
+
+We need to specify the Contract name for that Decorator so that it will know for which Template it is meant to Decorate.
+
+To recap, the Template we're trying to Decorate is the `StartupTemplate` and it requires a Decorator that adheres to this interface:
+
+```csharp
+public interface IStartupTemplateContract : ITemplateDecorator
+{
+    string ConfigureCode();
+}
+```
+
+It has a Full Namespace of: `MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract`
+
+So that is the value that we need to specify in the `Full TypeName` field for that Decorator.
```

Comment:
I'm thinking that we just have this in a NOTE and simplify it to something like:

```suggestion
> [!NOTE]
> This is the fully qualified name of the decorator interface which was created for your `StartupTemplate` in the [](xref:MakeTemplateDecoratorReady) article.
```

(As per one of my earlier comments, we could leave the square brackets empty to have the article name resolved automatically.)

Reply:
Perhaps we can change the wording but I do feel its important that people know what the contract looks like that they are working with. Perhaps we can say "We are attempting to Decorate the `StartupTemplate` and it requires that the Decorator adhere to this interface:"

Reply:
Because all we're doing at this point is entering values into the stereotype field, I don't think that what the interface looks like is relevant, _yet_. Later on in the article where we're implementing the interface, we could put in a code fence of what the interface looks like exactly.

Reply:
If order is the concern we can move it but I do feel that it is important that we keep it on the page

Reply:
I think we'll inline the entire interface as code further down in this article where the implementation needs to be done.

> [!NOTE]
> This is the fully qualified name of the decorator interface which was created for the `StartupTemplate` in the [](xref:MakeTemplateDecoratorReady) article, if you're implementing a different decorator interface then enter its fully qualified name instead.

Let me know how you feel about the suggested change or if you're happy feel free to commit it yourself.